File: robosys1.py
#!/usr/bin/python3
# SPDX-FileCopyrightText: 2025 Toma Misono
# SPDX-License-Identifier: BSD-3-Clause

a = int(input())

if a < 10000:
    print(a)
    exit()

if 10000 <= a < 100000000:
    quotient = a // 10000
    b = a % 10
    a //= 10
    c = a % 10
    a //= 10
    d = a % 10
    a //= 10
    e = a % 10
    a //= 10
    print(f"{quotient}万", end="")
    if e != 0:
        print(e, end="")
    if e != 0 or d != 0:
        print(d, end="")
    if e != 0 or d != 0 or c != 0:
        print(c, end="")
    if e != 0 or d != 0 or c != 0 or b != 0:
        print(b, end="")
    print()

File: robosys1.c
#!/usr/bin/python3
# SPDX-FileCopyrightText: 2025 Kaito Shima
# SPDX-License-Identifier: BSD-3-Clause

#include <stdio.h>

int main(void)
{
    int a;

    scanf("%d", &a);
    if (a < 10000)
        printf("%d\n", a);
    
    int quotient, b, c, d, e, f;
    if (10000 <= a && a < 100000000) {
        quotient = a / 10000;
        b = a % 10;
        a /= 10;
        c = a % 10;
        a /= 10;
        d = a % 10;
        a /= 10;
        e = a % 10;
        a /= 10;
        printf("%d万", quotient);
        if (e != 0)
            printf("%d", e);
        if (e != 0 || d != 0)
            printf("%d", d);
        if (e != 0 || d != 0 || c != 0)
            printf("%d", c);
        if (e != 0 || d != 0 || c != 0 || b != 0)
            printf("%d\n", b);
        
    }

    return 0;
}
